### PR TITLE
Fix #2822 : Add label AppVersionActivity 

### LINKED ISSUE #2822: [A11y] Add label for AppVersionActivity
## Task
Introduce label for `AppVersionActivity` as `App Version`.

Reference PR: https://github.com/oppia/oppia-android/pull/2750

When you create a PR for this, make sure you add test to check the functionality. Also, suggest working on this PR only if its your first or second PR to Oppia-Android project.

## General Explanation
When `Talkback` is on, whenever we click on any option which leads to new screen the reader will automatically read the `label` associated with that screen first. If in case label in not present then it will read app name i.e. `Oppia`.

### Example

https://user-images.githubusercontent.com/9396084/108906068-78b75080-7646-11eb-943e-212e29fc3e98.mp4

In this above video as soon as we double-tap `Battery`, new screen is opened. 
The reader first outputs `Battery` followed by `Navigate up, Button` and finally `Battery` again. In this the first `Battery` output is the `label` of this new screen which indicates the learner which screen is opened. The second `Battery` is because we selected the title in the screen and it's just reading that text.
So we need `labels` for various screens general which matches the title but if the title is dynamic then we need static label.

### PR DESCRIPTION
**Pull request by E13ctron**
## Explanation
Fixes #2822 : Add label AppVersionActivity

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [ ] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

File: .idea/markdown-navigator-enh.xml
Comment (FareesHussain) [inline .idea/markdown-navigator-enh.xml]: Please revert this file

File: .idea/markdown-navigator.xml
Comment (FareesHussain) [inline .idea/markdown-navigator.xml]: Please revert this file.

File: .idea/misc.xml
Comment (FareesHussain) [inline .idea/misc.xml]: Please revert this file

File: app/src/main/res/values/strings.xml
Comment (FareesHussain) [inline app/src/main/res/values/strings.xml]: Add this line below
https://github.com/oppia/oppia-android/blob/2d8930664b86a98b16ec1d0d98c6f2742ce13baa/app/src/main/res/values/strings.xml#L332

Similar to 
https://github.com/oppia/oppia-android/blob/2d8930664b86a98b16ec1d0d98c6f2742ce13baa/app/src/main/res/values/strings.xml#L302-L303

**Review by FareesHussain — CHANGES_REQUESTED** [review]
@Amar-2003 Thanks for the pr, left few comments do have a look.

According to the 4th point in [Instructions for making a code change](https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change)
- Update the PR description so that it includes `Fixes #2822`
- And before commiting make sure that you don't include the auto-generated code by Android-Studio, in your case includes the following files 
     - .idea/markdown-navigator-enh.xml
     - .idea/markdown-navigator.xml
     - .idea/misc.xml

**Comment by E13ctron** [comment]
Ok fine will do it @FareesHussain

**Comment by E13ctron** [comment]
> @Amar-2003 Thanks for the pr, left few comments do have a look.
> 
> According to the 4th point in [Instructions for making a code change](https://github.com/oppia/oppia-android/wiki#instructions-for-making-a-code-change)
> 
> * Update the PR description so that it includes `Fixes #2822`
> * And before commiting make sure that you don't include the auto-generated code by Android-Studio, in your case includes the following files
>   
>   * .idea/markdown-navigator-enh.xml
>   * .idea/markdown-navigator.xml
>   * .idea/misc.xml

Made the changes, please have a look and sorry for the inconvenience

File: app/src/main/res/values/strings.xml
Comment (FareesHussain) [inline app/src/main/res/values/strings.xml]: It would be better to use
`app_version_activity_label`

**Review by FareesHussain — CHANGES_REQUESTED** [review]
NIT fix, PTAL

File: app/src/main/res/values/strings.xml
Comment (E13ctron) [reply app/src/main/res/values/strings.xml]: > It would be better to use
> `app_version_activity_label`

made the change, thanks

**Comment by prayutsu** [comment]
@Amar-2003 Can you also add a test case to verify that the label is actually present?

**Comment by rt4914** [comment]
Will wait for approval from @prayutsu and @FareesHussain first. Thanks @Amar-2003

**Comment by rt4914** [comment]
@Amar-2003 If you check the original issue there is a reference PR mentioned. In that PR you will notice that there is test case which checks this functionality. So please add something similar for this PR too.

**Review by FareesHussain — APPROVED** [review]
LGTM, thanks

**Comment by E13ctron** [comment]
I don't know about adding test cases, so i will learn right away and do the neccessary changes

**Comment by prayutsu** [comment]
@Amar-2003 You can take a look at reference PR #2750.

**Comment by E13ctron** [comment]
in file AppVersionActivityTest, line 248 there is some error, should i mind it?? @prayutsu

**Comment by E13ctron** [comment]
> in file AppVersionActivityTest, line 248 there is some error, should i mind it?? @prayutsu

I am pushing my changes without considering that, so the line 248 changes to 263

**Comment by prayutsu** [comment]
@Amar-2003 PTAL, the lint tests are failing, please fix that so that we can review your PR.
Also, please choose a more meaningful branch name in your future PRs(it should not include your name/username and it should consist of only lowercase English letters and different words should be separated by `-`).

**Comment by E13ctron** [comment]
> @Amar-2003 PTAL, the lint tests are failing, please fix that so that we can review your PR.
> Also, please choose a more meaningful branch name in your future PRs(it should not include your name/username and it should consist of only lowercase English letters and different words should be separated by `-`).

The branch name doesn't include my user name.Lint test i will have a look :)

**Comment by prayutsu** [comment]
> > @Amar-2003 PTAL, the lint tests are failing, please fix that so that we can review your PR.
> > Also, please choose a more meaningful branch name in your future PRs(it should not include your name/username and it should consist of only lowercase English letters and different words should be separated by `-`).
> 
> The branch name doesn't include my user name.Lint test i will have a look :)

Ah, I always interpret it wrong, thanks for pointing it out.

**Comment by E13ctron** [comment]
> > > @Amar-2003 PTAL, the lint tests are failing, please fix that so that we can review your PR.
> > > Also, please choose a more meaningful branch name in your future PRs(it should not include your name/username and it should consist of only lowercase English letters and different words should be separated by `-`).
> > 
> > 
> > The branch name doesn't include my user name.Lint test i will have a look :)
> 
> Ah, I always interpret it wrong, thanks for pointing it out.

No Problem ;)

**Comment by prayutsu** [comment]
@Amar-2003 Also note that the error you are mentioning on the line number 248 will go away once you run the test(It applies for all our test files)

**Comment by E13ctron** [comment]
> @Amar-2003 Also note that the error you are mentioning on the line number 248 will go away once you run the test(It applies for all our test files)

actually I don't have a clear idea of running tests at present

**Comment by prayutsu** [comment]
You can go through our Instructions on [wiki](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests) to run the tests locally.

**Comment by E13ctron** [comment]
> You can go through our Instructions on [wiki](https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests) to run the tests locally.

Thanks

**Comment by E13ctron** [comment]
The oppia android repo too contains the same incorrect line 248 in file AppVersionActivityTest file, the same line causes error while running tests locally. @prayutsu

**Comment by rt4914** [comment]
Adding @prayutsu for review.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt
Comment (prayutsu) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt]: This test is incorrect and thus failing the CI checks.
Have a look at #2750 and try to correct it.

**Review by prayutsu — CHANGES_REQUESTED** [review]
@Amar-2003 PTAL, lef a comment.

**Comment by prayutsu** [comment]
> The oppia android repo too contains the same incorrect line 248 in file AppVersionActivityTest file, the same line causes error while running tests locally. @prayutsu

I didn't get what are you trying to say, can you please explain this more clearly?

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt
Comment (E13ctron) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt]: Thanks for mentioning it

**Comment by E13ctron** [comment]
> > The oppia android repo too contains the same incorrect line 248 in file AppVersionActivityTest file, the same line causes error while running tests locally. @prayutsu
> 
> I didn't get what are you trying to say, can you please explain this more clearly?

yeah sure, I got error on the line 263, so to check whether I made any mistake I just cloned the orginal repo and found that Android giving error for line 248 in file AppVersionActivityTest

**Comment by rt4914** [comment]
> AppVersionActivityTest

@Amar-2003 Android studio might be showing error at that line but if you run the tests they will work successfully. It shows error only because the project has not been built. If you build the project it will no show error.

**Comment by E13ctron** [comment]
> > AppVersionActivityTest
> 
> @Amar-2003 Android studio might be showing error at that line but if you run the tests they will work successfully. It shows error only because the project has not been built. If you build the project it will no show error.

ok, Thanks :)

File: app/src/main/res/values/strings.xml
Comment (rt4914) [inline app/src/main/res/values/strings.xml]: Rename this to `app_version_activity_title` for consistency.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt]: Use `val` instead of `var`

**Review by rt4914 — CHANGES_REQUESTED** [review]
Suggested change.

**Comment by E13ctron** [comment]
I have made the suggested changes, PTAL @prayutsu @rt4914

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt
Comment (prayutsu) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt]: It requires reformatting, the ')' should come in the next line.

**Review by prayutsu — CHANGES_REQUESTED** [review]
@Amar-2003 Just had a nit suggestion. PTAL.

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt
Comment (E13ctron) [reply app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt]: ok will do it

**Comment by E13ctron** [comment]
@prayutsu PTAL

**Comment by E13ctron** [comment]
Are there anymore corrections I should do?? @rt4914

File: app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt
Comment (prayutsu) [inline app/src/sharedTest/java/org/oppia/android/app/administratorcontrols/AppVersionActivityTest.kt]: We should avoid these change, only import `Locale` here.

**Review by prayutsu — CHANGES_REQUESTED** [review]
Left a comment. @Amar-2003 PTAL.

**Comment by E13ctron** [comment]
@prayutsu PTAL

**Review by prayutsu — APPROVED** [review]
LGTM, Thanks!

**Comment by E13ctron** [comment]
> LGTM, Thanks!

Thanks for patiently finding the issues and helping me to get through, this was my first PR to open source :)

**Comment by E13ctron** [comment]
@rt4914 PTAL

**Review by rt4914 — APPROVED** [review]
LGTM, thanks.

**Review by FareesHussain — APPROVED** [review]
LGTM, thanks.

**Comment by E13ctron** [comment]
@BenHenning PTAL

**Review by BenHenning — APPROVED** [review]
Thanks @Amar-2003! This LGTM.

**Review by E13ctron — COMMENTED** [review]
good

**Pull request merged**
